stonesoup/types/array.py

```diff
@@ -46,12 +77,24 @@ def __new__(cls, *args, **kwargs):
         return array.view(cls)
 
     def __array_wrap__(self, array):
-        return np.asarray(array)
+        return StateVector._cast(np.asarray(array))
```

Comment:
> Overall looks ok, but I have some comments:
> 
> * While having a play with the UKF test script you kindly shared last week, I noticed that after running the `get_measurement_prediction()` method, the bearing element in the `GaussianMeasurementPrediction.state_vector` returned by the UKF updater is not of `Bearing` type, while the same element returned by EKF is. I think this is to be expected, but I thought I'd bring it up just in case.
> * Seeing as #104 seems to overwrite/modify a large part of the changes introduced by this PR, it might be more appropriate to merge them into a single case. Surely @sdhiscocks should have the say on how to proceed, but the reason for suggesting this is that #104 looks a lot more polished and the concerns that occurred to me while reviewing this PR, seem to have been addressed in #104.

I'm not sure if the fixes in #104 will result in a Bearing type being returned by the UKF measurement prediction code. The type that is returned really depends on the operations that  occur within the UKF code. Pretty much the only operations that return a Bearing type are addition and subtraction. Multiplication and Division will just return a float.

I will modify #104 to supersede this Pull Request.